src/Strong_collapse/include/gudhi/Strong_collapse/Flag_complex_tower_assembler.h

```diff
@@ -0,0 +1,175 @@
+            renamed_vertices_.at(v) = current_rename_counter_;
+            current_rename_counter_++;
+          } else {
+            if (!outFile.empty()) {
+              myfile << filtration_value << " i " << renamed_vertices_.at(collapsed_to->second) << std::endl;
+              myfile << filtration_value << " c " << renamed_vertices_.at(v) << " "
+                     << renamed_vertices_.at(collapsed_to->second) << std::endl;
+            }
+            flag_filtration_.active_strong_expansion(renamed_vertices_.at(v), renamed_vertices_.at(collapsed_to->second),
+                                                     filtration_value);
+            renamed_vertices_.at(v) = current_rename_counter_;
+            current_rename_counter_++;
+          }
+          // If the vertex "collapsed_to->second" is not a member of mat_1, the contraction function will simply add and
+          // then collapse
+          mat_1.contraction(v, collapsed_to->second);
+        }
+      }
+
+      // The core K1c (mat_1) has gone through the transformation(re-labeling)/collapse and it is now a subcomplex of
+      // K2c, the remaining simplices need to be included
+      // Writing the inclusion of all remaining simplices...
+      for (const Edge& e : mat_2.all_edges()) {
+        auto u = std::get<0>(e);
+        auto v = std::get<1>(e);
+        if (!mat_1.membership(u)) {
+          flag_filtration_.insert_vertex(renamed_vertices_.at(u), filtration_value);
+          if (!outFile.empty()) {
+            myfile << filtration_value << " i";
+            myfile << " " << renamed_vertices_.at(u);
+            myfile << std::endl;
+          }
+          mat_1.insert_vertex(u, 1);
+        }
+
+        if (!mat_1.membership(v)) {
+          flag_filtration_.insert_vertex(renamed_vertices_.at(v), filtration_value);
+
+          if (!outFile.empty()) {
+            myfile << filtration_value << " i";
+            myfile << " " << renamed_vertices_.at(v);
+            myfile << std::endl;
+          }
+          mat_1.insert_vertex(v, 1);
+        }
+        if (!mat_1.membership(e)) {
+          flag_filtration_.insert_new_edges(renamed_vertices_.at(u), renamed_vertices_.at(v), filtration_value);
+
+          if (!outFile.empty()) {
+            myfile << filtration_value << " i";
+            myfile << " " << renamed_vertices_.at(u) << ", " << renamed_vertices_.at(v);
+            myfile << std::endl;
+          }
+          mat_1.insert_new_edges(u, v, 1);
+        }
+      }
+      if (!outFile.empty()) {
+        myfile << "# Tower updated for the additional subcomplex.\n";
+        myfile.close();
+      }
+    } else {
+      std::cerr << "Unable to open file " << outFile;
+      exit(-1);
+    }
+  }
+
+  typename Flag_complex_sparse_matrix<SimplicialComplex>::Distance_matrix distance_matrix() {
+    std::size_t non_zero_rw = flag_filtration_.num_vertices();
+    Filtration_value inf = std::numeric_limits<Filtration_value>::max();
```

Comment:
That's a misleading name. `max` is the largest **finite** value. For infinity, you can replace `max` with `infinity`. Then it depends which one you want...